### ISSUE TITLE
Externalizing DB settings 

### ISSUE THREAD

**Issue by Gargee** [Impetus/kundera#391]
Hi,
When i try to externalise my DB connection I am getting an error like:-
11 Sep 2013 15:13:24,964 ERROR localhost-startStop-1 com.impetus.kundera.service.HostConfiguration (HostConfiguration.java:110) - Host or port should not be null / port should be numeric.

My persistence.xml looks like this:-
<'persistence xmlns="http://java.sun.com/xml/ns/persistence"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://java.sun.com/xml/ns/persistence
    http://java.sun.com/xml/ns/persistence/persistence_2_0.xsd"
    version="2.0">
    <'persistence-unit name="cassandra_pu">
        <'provider>com.impetus.kundera.KunderaPersistence</provider>
        <'properties>

```
        <'property name="kundera.keyspace" value="EPCContent" />
        <'property name="kundera.dialect" value="cassandra" />
        <'property name="kundera.client.lookup.class"
            value="com.impetus.client.cassandra.pelops.PelopsClientFactory" />
        <'property name="kundera.cache.provider.class"
            value="com.impetus.kundera.cache.ehcache.EhCacheProvider" />
        <'property name="kundera.cache.config.resource" value="/ehcache-test.xml" />
        <'property name="kundera.client.property" value="kunderaClientConfig.xml" />
    <'/properties>
<'/persistence-unit>
```

<'/persistence>

my kunderaClientConfig.xml:-
<'?xml version="1.0" encoding="UTF-8"?>
<'clientProperties>
    <'datastores>
        <'dataStore>
            <'name>cassandra</name>
            <'connection>
                <'properties>
                    <'property name="cql.version" value="3.0.0"></property>
                    
                    <'property name="loadbalancing.policy" value="roundrobin"></property>
                    <'property name="retry" value="true"></property>
                    <'property name="retry.delay" value="1"></property>
                    <'property name="max.wait" value="3000"></property>
                <'/properties>

```
            <'servers>
                <'server>
                    <'host>
                        XYZ
                    <'/host>
                    <'port>
                        9160
                    <'/port>
                <'/server>
                <'/servers>         
        <'/connection>
    <'/dataStore>
<'/datastores>
```

<'/clientProperties>

I get this error when i try starting my server:-
11 Sep 2013 15:13:24,978 ERROR localhost-startStop-1 org.springframework.web.servlet.DispatcherServlet (FrameworkServlet.java:468) - Context initialization failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'emf-p' defined in ServletContext resource [/WEB-INF/springMVC.xml]: Invocation of init method failed; nested exception is java.lang.IllegalArgumentException: Host or port should not be null / port should be numeric.
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1486)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:524)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:461)
        at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:295)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:223)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:292)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:194)
        at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1117)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:922)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:479)
        at org.springframework.web.servlet.FrameworkServlet.configureAndRefreshWebApplicationContext(FrameworkServlet.java:652)
        at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:600)
        at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:666)
        at org.springframework.web.servlet.FrameworkServlet.initWebApplicationContext(FrameworkServlet.java:519)
        at org.springframework.web.servlet.FrameworkServlet.initServletBean(FrameworkServlet.java:460)
        at org.springframework.web.servlet.HttpServletBean.init(HttpServletBean.java:136)
        at javax.servlet.GenericServlet.init(GenericServlet.java:160)
        at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1280)
        at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1193)
        at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:1088)
        at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:5123)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5407)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1559)
        at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1549)
        at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
        at java.util.concurrent.FutureTask.run(FutureTask.java:166)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:724)
Caused by: java.lang.IllegalArgumentException: Host or port should not be null / port should be numeric.
        at com.impetus.kundera.service.HostConfiguration.onValidation(HostConfiguration.java:111)
        at com.impetus.client.cassandra.service.CassandraHostConfiguration.buildHosts(CassandraHostConfiguration.java:66)
        at com.impetus.kundera.service.HostConfiguration.buildHosts(HostConfiguration.java:89)
        at com.impetus.kundera.service.HostConfiguration.<init>(HostConfiguration.java:63)
        at com.impetus.client.cassandra.service.CassandraHostConfiguration.<init>(CassandraHostConfiguration.java:51)
        at com.impetus.client.cassandra.pelops.PelopsClientFactory.initialize(PelopsClientFactory.java:81)
        at com.impetus.kundera.loader.GenericClientFactory.load(GenericClientFactory.java:112)
        at com.impetus.kundera.configure.ClientMetadataBuilder.buildClientFactoryMetadata(ClientMetadataBuilder.java:52)
        at com.impetus.kundera.persistence.EntityManagerFactoryImpl.configureClientFactories(EntityManagerFactoryImpl.java:352)
        at com.impetus.kundera.persistence.EntityManagerFactoryImpl.<init>(EntityManagerFactoryImpl.java:116)
        at com.impetus.kundera.KunderaPersistence.createEntityManagerFactory(KunderaPersistence.java:83)
        at com.impetus.kundera.KunderaPersistence.createContainerEntityManagerFactory(KunderaPersistence.java:65)
        at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:288)
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:310)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1545)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1483)
        ... 29 more
Sep 11, 2013 3:13:25 PM org.apache.catalina.startup.HostConfig deployDirectory

**Comment by mevivs** [comment]
Hi,
if you change

```
 <'host>
                        XYZ
                    <'/host>
                    <'port>
                        9160
                    <'/port>
```

with 

```
 <host>XYZ</host>
 <port>9160</port>
```

Means, keeping start and end tag (host, port) in same line. It should work for you. It seems an issue related to parsing and i will get it fixed in 2.7.1 

-Vivek

**Comment by Gargee** [comment]
we need to externalise this file to make it easily changeable in any env.

**Comment by Gargee** [comment]
can we expect an externalisable kunderaconfig in the next release of kundera?

**Comment by impetus-opensource-admin** [comment]
Yes. We are working on it. Will post an update on this.

-Vivek

**Comment by mevivs** [comment]
This is fixed and releasing with 2.7.1
